### PR TITLE
Remove result types and clean up errors

### LINKED ISSUE #2063: Remove `pub type Result<T> = std::result::Result<T, Error>;`?

### LINKED ISSUE #2096: Put all Error enums in their own file
consistency and easier to find

### LINKED ISSUE #2099: Remove some error redundancy
Now that errors have been split, there is a lot of redundancy between enum and variants names.
Most of the time, variants shouldn't contain the enum name.

### PR DESCRIPTION
**Pull request by DaughterOfMars**
# Description of change

Accomplishes three things:
1. Remove result type defs (#2063)
2. Move errors to their own files (#2096)
3. Renames error variants to avoid verbosity (#2099)

### CODE REVIEW

File: sdk/src/client/secret/private_key.rs [hunk 1]
```diff
@@ -44,7 +44,7 @@ impl SecretManage for PrivateKeySecretManager {
         _address_indexes: Range<u32>,
         _options: impl Into<Option<GenerateAddressOptions>> + Send,
     ) -> Result<Vec<ed25519::PublicKey>, Self::Error> {
-        crate::client::Result::Ok(vec![self.0.public_key()])
+        Result::Ok(vec![self.0.public_key()])
```
Comment (thibault-martinez) [inline sdk/src/client/secret/private_key.rs]: Here and maybe more, `Result::` not needed

File: sdk/src/wallet/update.rs [hunk 2]
```diff
@@ -22,13 +22,9 @@ use crate::{
     wallet::events::types::{NewOutputEvent, SpentOutputEvent, TransactionInclusionEvent, WalletEvent},
 };
 
-impl<S: 'static + SecretManage> Wallet<S>
-where
-    crate::wallet::Error: From<S::Error>,
-    crate::client::Error: From<S::Error>,
```
Comment (thibault-martinez) [inline sdk/src/wallet/update.rs:28]: Can/should it be removed in more places?

Comment (thibault-martinez) [reply sdk/src/wallet/update.rs:28]: Ah I see you are removing in some places, what's the logic of removing or not?

Comment (DaughterOfMars) [reply sdk/src/wallet/update.rs:28]: just isn't required

Comment (DaughterOfMars) [reply sdk/src/wallet/update.rs:28]: it's needed whenever the secret manager is used basically

File: sdk/src/wallet/operations/syncing/addresses/output_ids/basic.rs [hunk 3]
```diff
@@ -5,18 +5,15 @@ use crate::{
...
-where
-    crate::wallet::Error: From<S::Error>,
-{
+impl<S: 'static + SecretManage> Wallet<S> {
     /// Returns output ids of basic outputs that have only the address unlock condition
     pub(crate) async fn get_basic_output_ids_with_address_unlock_condition_only(
         &self,
         bech32_address: impl ConvertTo<Bech32Address>,
-    ) -> crate::client::Result<Vec<OutputId>> {
+    ) -> Result<Vec<OutputId>, WalletError> {
```
Comment (thibault-martinez) [inline sdk/src/wallet/operations/syncing/addresses/output_ids/basic.rs:16]: Probably insignificant but we went from client Result to wallet Error, bit weird. I guess it was wrong before.

**Review by thibault-martinez — DISMISSED** [review]

File: bindings/core/src/method_handler/secret_manager.rs [hunk 4]
```diff
@@ -40,7 +40,7 @@ where
             let addresses = secret_manager
                 .generate_ed25519_addresses(coin_type, account_index, range, options)
                 .await
-                .map_err(iota_sdk::client::Error::from)?
+                .map_err(iota_sdk::client::ClientError::from)?
```
Comment (Thoralf-M) [inline bindings/core/src/method_handler/secret_manager.rs]: Just import iota_sdk::client::ClientError one at the top

Comment (thibault-martinez) [reply bindings/core/src/method_handler/secret_manager.rs]: done

**Review by Thoralf-M — APPROVED** [review]

**Review by thibault-martinez — APPROVED** [review]

**Pull request merged**